test/models/graphcast/test_graphcast_snmg.py

```diff
@@ -0,0 +1,171 @@
+    model_multi_gpu = GraphCastNet(
+        partition_size=world_size,
+        partition_group_name="graph_partition",
+        expect_partitioned_input=True,
+        produce_aggregated_output=False,
+        **model_kwds,
+    ).to(device=device, dtype=dtype)
+
+    model_multi_gpu = DistributedDataParallel(
+        model_multi_gpu, process_group=manager.group("graph_partition")
+    )
+
+    # initialize data
+    x_single_gpu = create_random_input(
+        input_res=model_kwds["input_res"], dim=model_kwds["input_dim_grid_nodes"]
+    ).to(device=device, dtype=dtype)
+    x_multi_gpu = x_single_gpu.detach().clone()
+    x_multi_gpu = (
+        x_multi_gpu[0]
+        .view(model_multi_gpu.module.input_dim_grid_nodes, -1)
+        .permute(1, 0)
+    )
+    x_multi_gpu = model_multi_gpu.module.g2m_graph.get_src_node_features_in_partition(
+        x_multi_gpu
+    )
+
+    # forward + backward passes
+    out_single_gpu = model_single_gpu(x_single_gpu)
+    loss = out_single_gpu.sum()
+    loss.backward()
+
+    out_multi_gpu = model_multi_gpu(x_multi_gpu)
+    loss = out_multi_gpu.sum()
+    loss.backward()
+
+    # numeric tolerances based on dtype
+    tolerances = {
+        torch.float32: (2e-3, 1e-6),
+        torch.bfloat16: (5e-2, 1e-3),
+        torch.float16: (5e-2, 1e-3),
+    }
+    atol, rtol = tolerances[dtype]
+
+    # compare forward, now fully materialize out_multi_gpu to faciliate comparison
+    _B, _C, _N = out_multi_gpu.shape
+    out_multi_gpu = out_multi_gpu.view(_C, _N).permute(1, 0)
+    out_multi_gpu = model_multi_gpu.module.m2g_graph.get_global_dst_node_features(
+        out_multi_gpu
+    )
+    out_multi_gpu = out_multi_gpu.permute(1, 0).view(out_single_gpu.shape)
+    diff = out_single_gpu - out_multi_gpu
+    diff = torch.abs(diff)
+    mask = diff > atol
+    assert torch.allclose(
+        out_single_gpu, out_multi_gpu, atol=atol, rtol=rtol
+    ), f"{mask.sum()} elements have diff > {atol} \n {out_single_gpu[mask]} \n {out_multi_gpu[mask]}"
+
+    # compare model gradients (ensure correctness of backward)
+    model_multi_gpu_parameters = list(model_multi_gpu.parameters())
+    for param_idx, param in enumerate(model_single_gpu.parameters()):
+        diff = param - model_multi_gpu_parameters[param_idx]
+        diff = torch.abs(diff)
+        mask = diff > atol
+        assert torch.allclose(
+            param, model_multi_gpu_parameters[param_idx], atol=atol, rtol=rtol
+        ), f"{mask.sum()} elements have diff > {atol} \n {param[mask]} \n {model_multi_gpu_parameters[param_idx][mask]}"
+
+    # cleanup distributed
+    del os.environ["RANK"]
+    del os.environ["LOCAL_RANK"]
+    del os.environ["WORLD_SIZE"]
+    del os.environ["MASTER_ADDR"]
+    del os.environ["MASTER_PORT"]
+
+    DistributedManager.cleanup()
+
+
+@pytest.mark.multigpu
+@pytest.mark.parametrize("dtype", [torch.float32, torch.float16, torch.bfloat16])
+@pytest.mark.parametrize("do_concat_trick", [False, True])
```

Comment:
Would it be worth to parameterize on checkpointing as well to ensure it works in conjunction with distributed graphs?